react-components/src/components/ResetPassword/useResetPassword.js

```diff
@@ -0,0 +1,41 @@
+/*******************************************************************************
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+import { useState, useCallback } from 'react';
+import { useMutation } from '@apollo/react-hooks';
+
+import MUTATION_RESET_PASSWORD from '../../queries/mutation_reset_password.graphql';
+
+const useResetPassword = () => {
+    const [status, setStatus] = useState('new');
+
+    const [resetPassword] = useMutation(MUTATION_RESET_PASSWORD);
+
+    const handleFormSubmit = useCallback(
+        async ({ email, token, password }) => {
```

Comment:
I think you should revalidate the email address here, and not just rely on the client-side validation which can easily be skipped, it's green-belt security ;-)

I'm also wondering if we can also use a CSRF token to make sure that nobody can just DDoS the form or do some kind of brute-force attack against Magento's reset password feature.

Reply:
> I think you should revalidate the email address here, and not just rely on the client-side validation which can easily be skipped, it's green-belt security ;-)

Validating the email again in JS is again client-side validation and can easily be skipped. No point in doing that twice ;)

> I'm also wondering if we can also use a CSRF token to make sure that nobody can just DDoS the form or do some kind of brute-force attack against Magento's reset password feature.

Good point. Currently we don't do this at all (login for example). I think this should be investigated in a separate issue.

Reply:
> Validating the email again in JS is again client-side validation and can easily be skipped. No point in doing that twice ;)

Good point, my brain was still in client-server mode ;-)

For CSRF I agree we could do this in a separate issue.